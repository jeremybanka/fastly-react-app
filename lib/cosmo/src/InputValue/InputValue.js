/* @flow */

import * as React from "react";
import { FormattedMessage } from "react-intl";
import { Text } from "../";

type TextType =
  | string
  | React.Element<typeof Text>
  | React.Element<typeof FormattedMessage>;

type InputValueProps = {
  htmlFor: string,
  children: TextType,
};
const InputValue = (props: InputValueProps) => {
  return (
    <Text
      as="label"
      htmlFor={props.htmlFor}
      letterSpacing="sm"
      color="grays.6"
      fontSize="xs"
      fontWeight="semibold"
    >
      {props.children}
    </Text>
  );
};

export default InputValue;
