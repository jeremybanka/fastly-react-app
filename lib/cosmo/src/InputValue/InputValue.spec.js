/* @flow */

import React from "react";
import { render } from "enzyme";
import InputValue from "./InputValue";

describe("InputValue test suite", () => {
  const component = <InputValue>Label</InputValue>;
  it("renders the stored snapshot", () => {
    expect(render(component)).toMatchSnapshot();
  });
});
