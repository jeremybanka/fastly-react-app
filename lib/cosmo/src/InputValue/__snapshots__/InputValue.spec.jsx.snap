// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputValue test suite renders the stored snapshot 1`] = `
<label
  class="Text-sc-6c1xd4-0 fXfEJU"
  color="grays.6"
  font-size="xs"
  font-weight="semibold"
  letter-spacing="sm"
>
  Label
</label>
`;
